doc/man7/migration_guide.pod

```diff
@@ -0,0 +1,1636 @@
+=pod
+
+=head1 NAME
+
+migration_guide - OpenSSL migration guide
+
+=head1 SYNOPSIS
+
+See the individual manual pages for details.
+
+=head1 DESCRIPTION
+
+In OpenSSL 3.0 the concept of a B<OSSL_LIB_CTX> library context has been added.
+A default NULL library context is used if no library context is created. This
+allows applications to work as they did in 1.1.1 with minimal changes.
+If the user creates an  B<OSSL_LIB_CTX> via OSSL_LIB_CTX_new() then many
+functions need to be changed to pass additional parameters to handle the
+library context.
```

Comment:
Any thoughts on describing if one chooses to not create an explicit library context to obtain the minimal changes when migrating to 3.0.0, what will he or she lose after switching to 3.0.0?

Reply:
The question then becomes - do we want to duplicate all the stuff that is listed in CHANGES.md? Which currently is the place that lists breaking changes. Maybe a reference is good enough??

Reply:
This file will presumably be distributed in Linux distro's, etc. The CHANGES.md file is not.  If you point to it, you'd have to point to the github version.  And how does this interact with @ddvo's #14741?

Reply:
There should be a discussion with @DDvO in regards to these changes tonight. Once that is figured out then we can do something from there...
Maintaining CHANGES and a migration guide with the same changes seems like a recipe for disaster :(

Reply:
Yeah, it's certainly not good to have much overlap between `CHANGES.md` and the migration guide.
To me it makes most sense to have a self-contained complete description in the migration guide
and have in CHANGES only rather abstract summaries, which point to the respective portion(s) of the migration guide.

Reply:
doc/man7/crypto.pod discusses the new concepts so I have referenced that document.

Reply:
I cannot find the reference to crypto.pod, neither a quick motivation in the migration guide why users should bother creating and using such a library context. Can you please point me to where this is done?

Reply:
Library context included and referenced now.